### ISSUE TITLE
test: intermittent waitNotes() failures with fast storage

### ISSUE THREAD

**Issue by ddiss** [ddiss/icyci#13]
Running icyci `go test` on a host with nvme storage frequently / always results in intermittent test failures, which aren't seen atop slower SSD storage. E.g.
```
--- FAIL: TestSignedTagUnsignedCommit (10.67s)
    icyci_test.go:146: imported GPG keypair at /tmp/TestSignedTagUnsignedCommit4008492271/001/gpg
    icyci_test.go:222: 
        [init]
                defaultBranch = main
        [user]
                name = icyCI test
                email = icyci@example.com
                signingKey = <icyci@example.com>
        [gpg]
                format = openpgp
    icyci_test.go:284: 81ef3553b7de80dd96d143a0d4bdf18554a715c9: unsigned commit: map[src_test.sh:echo "commitI: 0"]
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at 81ef3553b7de80dd96d143a0d4bdf18554a715c9
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at 81ef3553b7de80dd96d143a0d4bdf18554a715c9
    icyci_test.go:314: notes ready!
    icyci_test.go:284: daef2bdb5e80ac91948493c46e30862f5b98810c: unsigned commit: map[src_test.sh:echo "commitI: 1"]
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:295: updating remotes
    icyci_test.go:305: checking notes at daef2bdb5e80ac91948493c46e30862f5b98810c
    icyci_test.go:830: timeout while waiting for icyCI notes
```

I expect 33a5018 ("test: reduce eventLoop / notesWait poll intervals") to have influenced things here, but would like to find the root cause.

**Comment by Werkov** [comment]
OT but you may be touching the code, I noticed this was missing from the interval rework:
```diff
--- a/icyci.go
+++ b/icyci.go
@@ -591,7 +592,7 @@ func pollSource(retCh chan<- error, exitCh chan error, sourceDir string,
                        goto err_out
 
                case <-pollTimer.C:
-                       pollTimer.Reset(time.Second * time.Duration(pollInterval))
+                       pollTimer.Reset(pollInterval)
 
                        err = pollFetch(sourceDir, branch)
                        if err != nil {
```

which likely works due to some typing luck in go (`Duration` from `Duration`).

**Comment by ddiss** [comment]
Good catch! wow, yeah I'm somewhat surprised this doesn't throw type errors.

**Comment by ddiss** [comment]
@Werkov Interestingly/amazingly, with a pollInterval of `300ms`, the incorrect `Reset` duration `time.Second * time.Duration(pollInterval)` evaluates to `83333h20m0s`. I won't go down the rabbit hole of trying to understand where that value comes from for now. Good thing is that your patch sees the unit tests pass quite consistently - thank you!

One reason why we didn't see this failure so much with larger poll intervals was that we'd normally detect the source change on the first fetch.

**Comment by ddiss** [comment]
Fixed via 578f3bf4c529f401ff492bc23ce14e2f8139df56 .

**Comment by Werkov** [comment]
> evaluates to 83333h20m0s. I won't go down the rabbit hole of trying to understand where that value comes from for now. 

I'll gladly peek -- it is 300e6 seconds (megaseconds), possibly parsed from `ms` (although correct SI prefix would be `Ms`). I thought it worked in my setup with default `60s` interval even on re-check.
